pkgs/desktops/gnome-3/devtools/gnome-builder/default.nix

```diff
@@ -0,0 +1,78 @@
+{stdenv
+, clang
+, defaultIconTheme
+, devhelp
+, fetchgit
+, gettext
+, gnome_themes_standard
+, gspell
+, gtk3
+, gtksourceview
+, hicolor_icon_theme
+, json_glib
+, jsonrpc_glib
+, libdazzle
+, libgit2-glib
+, libpeas
+, libxml2
+, llvmPackages
+, meson
+, ninja
+, pcre
+, pkgconfig
+, pkgs
+, python3
+, python3Packages
+, sysprof
+, template_glib
+, vala
+, vte
+, webkitgtk
+, wrapGAppsHook
+, ...}:
+
+stdenv.mkDerivation rec {
+  version = "3.27.2";
+  name = "gnome-builder-${version}";
+  nativeBuildInputs = [ gettext meson ninja pkgconfig python3 wrapGAppsHook ];
+  buildInputs = [
+    defaultIconTheme devhelp gspell gtk3 gtksourceview json_glib jsonrpc_glib
+    libdazzle libgit2-glib libpeas libxml2 llvmPackages.clang llvmPackages.llvm
+    pcre sysprof template_glib vala vte webkitgtk
+  ];
+
+  src = fetchgit {
+    url = git://git.gnome.org/gnome-builder;
+    rev = version;
+    sha256 = "0rna3wwpysba1kx9d9j7i97s4zslw0nya3p7ypd674y5cj4dla2j";
+  };
+
+  mesonFlags = "-D with_flatpak=false";
+
+  patches = [ ./patch_libide.patch ];
+
+  postPatch = ''
+    patchShebangs build-aux/meson/post_install.py
+    sed -e "s,^clang_include = ret.stdout().strip(),clang_include = '${clang.cc}/include/'," \
+        -e "s,^clang_libdir = ret.stdout().strip().split(' '),clang_libdir = '${clang.cc}/lib/'," \
+       -i src/plugins/clang/meson.build
+  '';
+
+  DESTDIR="/";
+
+  pythonPath = with python3Packages; makePythonPath [ pygobject3 ];
+  preFixup = ''
+    glib-compile-schemas $out/share/glib-2.0/schemas
```

Comment:
If you move this to `postInstall`, [glib setup hook](https://github.com/NixOS/nixpkgs/blob/5c66d653a267ecbddcc0e7d0025564006105dc51/pkgs/development/libraries/glib/setup-hook.sh) will take care of setting `GSETTINGS_SCHEMAS_PATH`.

Reply:
Thanks, changed that.